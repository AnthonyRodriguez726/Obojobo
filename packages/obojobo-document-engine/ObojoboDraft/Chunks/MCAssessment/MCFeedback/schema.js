import { Block } from 'slate'
import { CHILD_TYPE_INVALID } from 'slate-schema-violations'

const TEXT_NODE = 'ObojoboDraft.Chunks.Text'

const schema = {
	blocks: {
		'ObojoboDraft.Chunks.MCAssessment.MCFeedback': {
			nodes: [{ match: [{ type: 'oboeditor.component' }], min: 1 }],
			normalize: (editor, error) => {
				const { node, child, index } = error
				switch (error.code) {
					case 'child_min_invalid': {
						const block = Block.create({
							object: 'block',
							type: 'oboeditor.component',
							nodes: [
								{
									object: 'block',
									type: TEXT_NODE
								}
							]
						})
						return editor.insertNodeByKey(node.key, index, block)
					}
					case CHILD_TYPE_INVALID: {
						const block = Block.fromJSON({
							object: 'block',
							type: 'oboeditor.component',
							nodes: [
								{
									object: 'block',
									type: TEXT_NODE
								}
							]
						})
						return editor.withoutNormalizing(c => {
							c.removeNodeByKey(child.key)
							return c.insertNodeByKey(node.key, index, block)
						})
					}
				}
			}
		}
	}
}

export default schema
