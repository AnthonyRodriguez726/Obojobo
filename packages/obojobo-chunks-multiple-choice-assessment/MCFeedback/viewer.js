import Common from 'obojobo-document-engine/src/scripts/common/index'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.MCAssessment.MCFeedback', {
	adapter: null,
	componentClass: ViewerComponent,
	type: 'chunk'
})
