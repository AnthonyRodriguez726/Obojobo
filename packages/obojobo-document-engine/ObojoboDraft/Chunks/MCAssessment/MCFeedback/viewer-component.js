import './viewer-component.scss'

import Common from 'Common'

let { OboComponent } = Common.components

export default props => (
	<OboComponent
		model={props.model}
		moduleData={props.moduleData}
		className={`obojobo-draft--chunks--mc-assessment--mc-feedback${
			props.model.parent.modelState.score === 100
				? ' is-correct-feedback'
				: ' is-not-correct-feedback'
		}`}
		data-choice-label={props.label}
	>
		{props.model.children.models.map((child, index) => {
			let Component = child.getComponentClass()
			return <Component key={child.get('id')} model={child} moduleData={props.moduleData} />
		})}
	</OboComponent>
)
