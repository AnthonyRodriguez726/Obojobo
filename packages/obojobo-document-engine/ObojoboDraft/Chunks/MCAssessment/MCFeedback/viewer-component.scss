@import '~styles/includes';

.obojobo-draft--chunks--mc-assessment--mc-feedback {
	// @keyframes obojobo-draft--chunks--mc-assessment--mc-feedback--bounce {
	// 	0% {
	// 		//transform: scale(0);
	// 		transform: translate(-100vw);
	// 	}

	// 	// 50% {
	// 	// 	transform: scale(1.1);
	// 	// }

	// 	// 80% {
	// 	// 	transform: scale(0.9);
	// 	// }

	// 	100% {
	// 		//transform: scale(1);
	// 		transform: translate(0);
	// 	}
	// }

	font-size: 0.8em;
	padding-left: $dimension-padding * 0.2333333333;
	padding-right: $dimension-padding * 0.2333333333;
	// animation: obojobo-draft--chunks--mc-assessment--mc-feedback--bounce 1s;
}
