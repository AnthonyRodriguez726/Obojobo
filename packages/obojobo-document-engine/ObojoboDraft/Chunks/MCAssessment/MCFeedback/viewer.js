import Common from 'Common'

import ViewerComponent from './viewer-component'

let SelectionHandler = Common.chunk.textChunk.TextGroupSelectionHandler

Common.Store.registerModel('ObojoboDraft.Chunks.MCAssessment.MCFeedback', {
	type: 'chunk',
	adapter: null,
	componentClass: ViewerComponent,
	selectionHandler: new SelectionHandler()
})
