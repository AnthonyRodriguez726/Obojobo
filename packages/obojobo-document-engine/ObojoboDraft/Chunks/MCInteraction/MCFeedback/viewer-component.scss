@import '~styles/includes';

.obojobo-draft--chunks--mc-interaction--mc-feedback
{
	font-size: 0.8em;
	padding-left: $dimension-padding * 0.2333333333;
	padding-right: $dimension-padding * 0.2333333333;
}