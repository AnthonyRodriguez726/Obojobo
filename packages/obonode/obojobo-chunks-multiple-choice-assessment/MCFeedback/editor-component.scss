@import '~styles/includes';

.obojobo-draft--chunks--mc-assessment--mc-feedback {
	margin-top: 2em;
	padding-top: 2em;
	padding-bottom: 2em;
	pointer-events: all;
	background: $color-bg2;
}
