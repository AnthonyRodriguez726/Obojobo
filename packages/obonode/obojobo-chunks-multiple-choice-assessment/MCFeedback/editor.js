import React from 'react'
import Common from 'obojobo-document-engine/src/scripts/common'

import Node from './editor-component'
import Schema from './schema'
import Converter from './converter'

const MCFEEDBACK_NODE = 'ObojoboDraft.Chunks.MCAssessment.MCFeedback'

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case MCFEEDBACK_NODE:
				return <Node {...props} {...props.attributes} />
		}
	},
	schema: Schema
}

Common.Registry.registerModel('ObojoboDraft.Chunks.MCAssessment.MCFeedback', {
	name: 'Multiple Choice Feedback',
	isInsertable: false,
	slateToObo: Converter.slateToObo,
	oboToSlate: Converter.oboToSlate,
	plugins
})

const MCFeedback = {
	components: {
		Node
	},
	helpers: {
		slateToObo: Converter.slateToObo,
		oboToSlate: Converter.oboToSlate
	},
	plugins
}

export default MCFeedback
