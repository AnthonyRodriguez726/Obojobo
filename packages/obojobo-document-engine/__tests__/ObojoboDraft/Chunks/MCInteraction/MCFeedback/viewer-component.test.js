import React from 'react'
import renderer from 'react-test-renderer'
import MCFeedback from '../../../../../ObojoboDraft/Chunks/MCInteraction/MCFeedback/viewer-component'
import OboModel from '../../../../../__mocks__/_obo-model-with-chunks'

describe('MCFeedback', () => {
	let moduleData = {
		focusState: {}
	}

	test('MCFeedback for a correct item', () => {
		let model = OboModel.create({
			id: 'parent',
			type: 'ObojoboDraft.Chunks.MCInteraction.MCChoice',
			content: {
				score: 100
			},
			children: [
				{
					id: 'self',
					type: 'ObojoboDraft.Chunks.MCInteraction.MCFeedback'
				}
			]
		})
		const component = renderer.create(
			<MCFeedback moduleData={moduleData} model={OboModel.models.self} />
		)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('MCFeedback for an incorrect item', () => {
		let model = OboModel.create({
			id: 'parent',
			type: 'ObojoboDraft.Chunks.MCInteraction.MCChoice',
			content: {
				score: 0
			},
			children: [
				{
					id: 'self',
					type: 'ObojoboDraft.Chunks.MCInteraction.MCFeedback'
				}
			]
		})
		const component = renderer.create(
			<MCFeedback moduleData={moduleData} model={OboModel.models.self} />
		)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
