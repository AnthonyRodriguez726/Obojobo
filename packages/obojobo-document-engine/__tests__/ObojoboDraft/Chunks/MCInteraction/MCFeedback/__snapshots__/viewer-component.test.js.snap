// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MCFeedback MCFeedback for a correct item 1`] = `
<div
  className="component obojobo-draft--chunks--mc-interaction--mc-feedback is-correct-feedback"
  data-choice-label={undefined}
  data-focussed={false}
  data-id="self"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.MCInteraction.MCFeedback"
  id="obo-self"
/>
`;

exports[`MCFeedback MCFeedback for an incorrect item 1`] = `
<div
  className="component obojobo-draft--chunks--mc-interaction--mc-feedback is-incorrect-feedback"
  data-choice-label={undefined}
  data-focussed={false}
  data-id="self"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.MCInteraction.MCFeedback"
  id="obo-self"
/>
`;
